src/main/java/com/google/cloud/spanner/pgadapter/utils/Converter.java

```diff
@@ -14,120 +14,26 @@
 
 package com.google.cloud.spanner.pgadapter.utils;
 
+import com.google.cloud.spanner.ResultSet;
 import com.google.cloud.spanner.pgadapter.ProxyServer.DataFormat;
-import com.google.cloud.spanner.pgadapter.metadata.SQLMetadata;
 import com.google.cloud.spanner.pgadapter.parsers.Parser;
 import com.google.common.base.Preconditions;
-import com.google.common.collect.HashMultimap;
-import com.google.common.collect.SetMultimap;
-import java.sql.ResultSet;
-import java.sql.ResultSetMetaData;
-import java.sql.SQLException;
 
 /** Utility class for converting between generic PostgreSQL conversions. */
 public class Converter {
-
-  /**
-   * PostgreSQL parameters occur as $\\d+, whereas JDBC expects a question mark. For now, doing a
-   * simple iteration to correct. Taking into account both quoted strings and escape sequences.
-   *
-   * @param sql The PostgreSQL String.
-   * @return A {@link SQLMetadata} object containing both the corrected SQL String as well as the
-   *     number of parameters iterated.
-   */
-  public static SQLMetadata toJDBCParams(String sql) {
```

Comment:
This is no longer needed, as we can feed the SQL string with the PostgreSQL-style parameters directly to the Connection API.

Reply:
Nice